otter/convergence/transforming.py

```diff
@@ -0,0 +1,117 @@
+"""
+Functions for transforming collections of steps.
+
+There are a few classes of transformations here:
+
+- optimizing (converting multiple steps into one for the purposes of reducing
+  API roundtrips)
+- limiting (by truncating the number of steps we take)
+- throttling (by wrapping multiple steps with one that gradually executes them)
+"""
```

Comment:
yes, there are many different ways to categorize "roles" -- the one I used here is the role of "transformation of steps after the general plan has been calculated".

I do not think any of the code here should stay in plan.py. It's better to err on the side of smaller modules than bigger ones. Would you like me to create "optimizing.py", "throttling.py", and "limiting.py"?